irohad/ametsuchi/impl/postgres_command_executor.cpp

```diff
@@ -0,0 +1,792 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ametsuchi/impl/postgres_command_executor.hpp"
+
+#include <boost/format.hpp>
+
+#include "ametsuchi/impl/postgres_wsv_common.hpp"
+#include "backend/protobuf/permissions.hpp"
+#include "interfaces/commands/add_asset_quantity.hpp"
+#include "interfaces/commands/add_peer.hpp"
+#include "interfaces/commands/add_signatory.hpp"
+#include "interfaces/commands/append_role.hpp"
+#include "interfaces/commands/create_account.hpp"
+#include "interfaces/commands/create_asset.hpp"
+#include "interfaces/commands/create_domain.hpp"
+#include "interfaces/commands/create_role.hpp"
+#include "interfaces/commands/detach_role.hpp"
+#include "interfaces/commands/grant_permission.hpp"
+#include "interfaces/commands/remove_signatory.hpp"
+#include "interfaces/commands/revoke_permission.hpp"
+#include "interfaces/commands/set_account_detail.hpp"
+#include "interfaces/commands/set_quorum.hpp"
+#include "interfaces/commands/subtract_asset_quantity.hpp"
+#include "interfaces/commands/transfer_asset.hpp"
+#include "interfaces/common_objects/types.hpp"
+
+iroha::expected::Error<iroha::ametsuchi::CommandError> makeCommandError(
+    const std::string &error_message,
+    const std::string &command_name) noexcept {
+  return iroha::expected::makeError(
+      iroha::ametsuchi::CommandError{command_name, error_message});
+}
+
+/**
+ * Transforms soci statement to CommandResult,
+ * which will have error message generated by error_generator
+ * appended to error received from given result
+ * @param result which can be received by calling execute_
+ * @param error_generator function which must generate error message
+ * to be used as a return error.
+ * Function is passed instead of string to avoid overhead of string
+ * construction in successful case.
+ * @return CommandResult with combined error message
+ * in case of result contains error
+ */
+template <typename Function>
+iroha::ametsuchi::CommandResult makeCommandResult(
+    soci::statement &st,
+    std::string command_name,
```

Comment:
const char * is passed here. It is not possible to convert char * to std::string &

Reply:
To make reference here, it is necessary to construct std::string outside this method. But since char * is always passed, I decided to construct string here